src/main/java/jeresources/profiling/Profiler.java

```diff
@@ -35,20 +37,44 @@ public void run() {
         LogHelper.warn("There will be messages about world gen lag during the profiling, you can ignore these as that is what you get when profiling.");
         if (!allWorlds) {
             WorldServer world = (WorldServer) this.sender.getEntityWorld();
-            profileWorld(world);
-        } else {
-            for (WorldServer world : DimensionManager.getWorlds()) {
-                profileWorld(world);
-            }
+            Entity sendEnt = sender.getCommandSenderEntity();
+    		int dimId = sendEnt.dimension;
+    		if(world == null) {
+				DimensionManager.initDimension(dimId);
+				world = DimensionManager.getWorld(dimId);
+    		}				
+    		profileWorld(world, dimId);
+        } else {		
+        	Integer[] dimIds = DimensionManager.getStaticDimensionIDs();
+        	for (int i = 0; i < dimIds.length; i++) {
+        		int dimId = dimIds[i];
+        		WorldServer world = DimensionManager.getWorld(dimId);
+        		if(world == null) {
+					DimensionManager.initDimension(dimId);
+					world = DimensionManager.getWorld(dimId);
+        		}				
+        		profileWorld(world, dimId);
+        	}
         }
 
         writeData();
 
         this.timer.complete();
     }
 
-    private void profileWorld(final WorldServer worldServer) {
-        int dimId = worldServer.provider.getDimensionType().getId();
+    private void profileWorld(final WorldServer worldServer, int dimId) {
+    	String msg2 = "Inspecting dimension ";
+    	try { 
+    			msg2 += worldServer.provider.getDimensionType().getName() + ": "; 
+    	}
+    	catch (Exception x)
```

Comment:
A catch all is about the dirtiest thing you can do in Java. At least use the expected exception.

Reply:
I know it is dirty, but it was more important for me to have it keep running and not stop.  If it was really going to be a show stopper, the problem would occur later anyway.